storage/src/cache/worker.rs

```diff
@@ -238,15 +238,17 @@ impl AsyncWorkerMgr {
 
     async fn handle_prefetch_requests(mgr: Arc<AsyncWorkerMgr>, rt: &Runtime) {
         // Max 2 active requests per thread.
-        mgr.prefetch_sema.add_permits(2);
+        mgr.prefetch_sema.add_permits(1);
 
         while let Ok(msg) = mgr.prefetch_channel.recv().await {
             mgr.handle_prefetch_rate_limit(&msg).await;
             let mgr2 = mgr.clone();
 
             match msg {
                 AsyncPrefetchMessage::BlobPrefetch(state, blob_cache, offset, size) => {
-                    let _ = mgr2.prefetch_sema.acquire().await;
+                    let token = Semaphore::acquire_owned(mgr2.prefetch_sema.clone())
```

Comment:
Could you help explain what is the specific bug with semaphore?

Reply:
> Could you help explain what is the specific bug with semaphore?

My 2 cents.
Instead of `add_permits(1)`,  use the existing 2 permit.

Reply:
Dropping the object returned by `Semaphore::acquire()` will increase the semaphore, so there are two issues here:
1) drop the returned object too early, it should be dropped by the async task instead.
2) shouldn't call `add_permits()`.